src/workers/postUpdated.ts

```diff
@@ -0,0 +1,68 @@
+import { messageToJson, Worker } from './worker';
+import { ArticlePost, SharePost } from '../entity';
+
+interface Data extends ArticlePost {
+  updated_at: Date;
+}
+
+const worker: Worker = {
+  subscription: 'api.post-updated',
+  handler: async (message, con, logger): Promise<void> => {
+    const data: Data = messageToJson(message);
+    try {
+      const { id, updated_at } = data;
+      const updatedDate = new Date(updated_at);
+      await con.transaction(async (entityManager) => {
+        const databasePost = await entityManager
+          .getRepository(ArticlePost)
+          .findOneBy({ id });
+
+        if (
+          !databasePost ||
+          databasePost.metadataChangedAt.toISOString() >=
+            updatedDate.toISOString()
+        ) {
+          return;
+        }
+
+        const becomesVisible = !databasePost?.visible && !!data?.title?.length;
+
+        await entityManager.getRepository(ArticlePost).update(
+          { id: databasePost.id },
+          {
+            summary: data?.summary,
+            description: data?.description,
+            readTime: data?.readTime,
+            toc: data?.toc,
+            keywords: data?.keywords,
```

Comment:
There's a whole logic for the keywords update.
I suggest having a look at our existing `newPost` endpoint to make sure we don't miss anything.
Essentially, this worker will take charge after we finish the migration, so we need to ensure everything is here.